### PR TITLE
Fixed #28052 -- Prevented dropping Meta.indexes when changing db_index to False.

### PR DESCRIPTION
**Pull request by MarkusH**
https://code.djangoproject.com/ticket/28052

### CODE REVIEW

**Comment by timgraham** [comment]
(is this ready for review and you forgot to update the ticket?)

**Comment by MarkusH** [comment]
Thanks for the ping, @timgraham . Updated.

**Comment by MarkusH** [comment]
@mjtamlyn @Ian-Foote I'd love a second :tm: pair of :eyes: here :)

File: tests/schema/tests.py [hunk 1]
```diff
@@ -1633,6 +1634,43 @@ def test_add_remove_index(self):
...
+        new_field.set_attributes_from_name('name')
+        with connection.schema_editor() as editor:
+            editor.alter_field(AuthorWithIndexedName, old_field, new_field, strict=True)
+        self.assertIn(
+            'author_name_idx',
+            self.get_constraints(AuthorWithIndexedName._meta.db_table)
+        )
+
+        # Reset the indexes attribute
+        AuthorWithIndexedName._meta.indexes = []
```
Comment (Ian-Foote) [inline tests/schema/tests.py]: Do we want to do this in a `finally:` clause?

Comment (MarkusH) [inline tests/schema/tests.py]: Good point. Changed.

**Pull request merged**